interpreter/prune.go

```diff
@@ -341,6 +341,12 @@ func (p *astPruner) prune(node *exprpb.Expr) (*exprpb.Expr, bool) {
 		}
 	}
 	if macro, found := p.macroCalls[node.GetId()]; found {
+		// Ensure that intermediate values for the comprehension are cleared during pruning
+		compre := node.GetComprehensionExpr()
+		if compre != nil {
+			visit(macro, clearIterVarVisitor(compre.IterVar, p.state))
+			visit(macro, clearIterVarVisitor(compre.AccuVar, p.state))
```

Comment:
Does the accumulation variable actually need to be cleared here? If so, could we add a test for this

Reply:
Oh, no, you're right. We can skip the accumulation variable clearing since it never appears in the macro call AST